test/comparator/test_file_set_comparator.py

```diff
@@ -300,6 +300,66 @@ def test_resources_existing_pattern_removal(self):
             "foo.proto",
         )
 
+    def test_resources_existing_patterns_reorder_in_place(self):
+        options_original = make_file_options_resource_definition(
+            resource_type=".example.v1.Bar",
+            resource_patterns=["bars/{bar}", "foos/{foo}"],
+        )
+        file_pb2 = make_file_pb2(
+            name="foo.proto", package=".example.v1", options=options_original
+        )
+        file_set_original = make_file_set(files=[file_pb2])
+
+        options_update = make_file_options_resource_definition(
+            resource_type=".example.v1.Bar",
+            resource_patterns=["foos/{foo}", "bars/{bar}"],
+        )
+        file_pb2 = make_file_pb2(
+            name="foo.proto", package=".example.v1", options=options_update
+        )
+        file_set_update = make_file_set(files=[file_pb2])
+
+        FileSetComparator(
+            file_set_original, file_set_update, self.finding_container
+        ).compare()
+        finding = self.finding_container.get_all_findings()[0]
+        self.assertEqual(finding.change_type.name, "MAJOR")
+        self.assertEqual(finding.category.name, "RESOURCE_PATTERN_REORDER")
```

Comment:
Could we also include a test case to check that inserting a pattern at the end of the list doesn't result in `finding.change_type.name` being `MAJOR`?